simphony_ui/tests/test_ui.py

```diff
@@ -24,29 +27,65 @@ def mock_run_calc(global_settings, openfoam_settings,
             progress_callback(100)
             return openfoam_wrapper, liggghts_wrapper
 
-        with mock.patch('simphony_ui.ui.run_calc') as mock_run:
+        dataset2cudssource_target = 'simphony_ui.ui.dataset2cudssource'
+
+        def mock_dataset2cudssource(*args, **kwargs):
+            return CUDSSource()
+
+        default_module_target = 'simphony_ui.ui.default_module'
+
+        def mock_default_module(*args, **kwargs):
+            return [1]
+
+        add_module_target = \
+            'simphony_ui.ui.mayavi.tools.mlab_scene_model.Engine.add_module'
+
+        def mock_add_module(module):
+            print module
+
+        with mock.patch(run_calc_target) as mock_run:
             mock_run.side_effect = mock_run_calc
 
-            self.assertIsNone(self.application.openfoam_wrapper)
-            self.assertIsNone(self.application.liggghts_wrapper)
+            with mock.patch(dataset2cudssource_target) as mock_cuds:
+                mock_cuds.side_effect = mock_dataset2cudssource
 
-            with self.event_loop_until_condition(
-                    lambda: self.application.openfoam_wrapper is not None,
-                    timeout=20
-            ):
-                self.application.run_calc()
+                with mock.patch(default_module_target) as mock_def:
+                    mock_def.side_effect = mock_default_module
 
-            self.assertEqual(
-                self.application.openfoam_wrapper,
-                openfoam_wrapper
-            )
-            self.assertEqual(
-                self.application.liggghts_wrapper,
-                liggghts_wrapper
-            )
+                    with mock.patch(add_module_target) as mock_add:
```

Comment:
You can write multiple context managers in a single with statement. This will reduce the amount of indentation
```
with A: 
  with B:
    with C:
```
becomes 
```
with A, B, C
```